Add Mixtral8x7b Model to T3000 demos

mtairum opened the pull request:
Initial version of Mixtral8x7b running on T3000 (8 chips), including all relevant tests added to multichip CI.

Backup branch with the full commit history: `mixtral8x7b-wh`

FYI:
@yieldthought 
@sraizada-tt 
@xuncaiTT

mtairum committed:
#5337: Add Mixtral8x7b T3000  to model demos and its respective CI demos.

mtairum commented:
We've added smaller tests to the `Multichip unit tests`workflow. All of these are 1 layer (if applicable), and are running on 8 devices. Execution should be around a couple of minutes for each, a bit longer for the decoder/model tests.

```
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_attention.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_mlp.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_rms_norm.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_embedding.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_moe.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_decoder.py
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py::test_mixtral_model_inference[1-1]
```

We've also added a larger test, full model (32L), 10 iterations to the `nightly multichip tests`.
```
pytest models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py::test_mixtral_model_inference[10-32]
```

@tapspatel Please let us know if these tests fit the assigned pipelines.

tapspatel commented:
looks good for the functional tests,

for multi chip perf pipeline: check this: https://github.com/tenstorrent/tt-metal/actions/workflows/multi-device-perf-models.yaml

tt-rkim commented on tests/scripts/run_performance.sh:
```diff
@@ -53,6 +53,11 @@ run_perf_models_llm_javelin_multi_device() {
     env pytest models/demos/falcon7b/tests -m $test_marker
+
+    # Mistral8x7b weights and cached files
+    export MIXTRAL_CKPT_DIR=/mnt/MLPerf/tt_dnn-models/Mistral/Mixtral-8x7B-v0.1
```
@tapspatel @TT-billteng I think we need to stop putting env vars in scripts like this.

Should we instruct Miguel to check for `os.environ["CI"] == "true"` and set these in the test itself?

mtairum replied:
Thanks for the warning.
I will update the perf test with this.

Should we update all of our tests to include this verification?

tt-rkim replied:
We want to provide something more robust for model test writers to use so it's cleaner and clearer that it's on CI.

But this is how we would check it's on CI.

mtairum replied:
Just pushed a commit with this change for all tests. Re-running the 3 pipelines now.

TT-billteng replied:
feel like this is a model cache and we should have this somewhere as a pythorn/pytest fixture

mtairum commented:
Squashed all commits and fixed all the above CI issues.
I've seen them pass in previous non-squashed version, but re-running them now on this clean commit to be double sure.
